src/m365/spo/commands/term/term-list.ts

```diff
@@ -108,36 +115,105 @@ class SpoTermListCommand extends SpoCommand {
-      };
-
-      const processQuery: string = await request.post(requestOptions);
-      const json: ClientSvcResponse = JSON.parse(processQuery);
-      const response: ClientSvcResponseContents = json[0];
-      if (response.ErrorInfo) {
-        throw response.ErrorInfo.ErrorMessage;
+      const result = await this.executeCsomCall(data, spoAdminUrl, res);
+      const terms: Term[] = [];
+      if (result._Child_Items_ && result._Child_Items_.length > 0) {
+        for (const term of result._Child_Items_) {
+          this.setTermDetails(term);
+          terms.push(term);
+          if (args.options.includeChildTerms && term.TermsCount > 0) {
+            await this.getChildTerms(spoAdminUrl, res, term, terms);
+          }
+        }
       }
 
-      const result: TermCollection = json[json.length - 1];
-      if (result._Child_Items_ && result._Child_Items_.length > 0) {
-        result._Child_Items_.forEach(t => {
-          t.CreatedDate = new Date(Number(t.CreatedDate.replace('/Date(', '').replace(')/', ''))).toISOString();
-          t.Id = t.Id.replace('/Guid(', '').replace(')/', '');
-          t.LastModifiedDate = new Date(Number(t.LastModifiedDate.replace('/Date(', '').replace(')/', ''))).toISOString();
-        });
-        logger.log(result._Child_Items_);
+      if (!args.options.output || args.options.output === 'json') {
+        logger.log(terms);
+      }
+      else {
+        // Converted to text friendly output
+        if (!args.options.includeChildTerms) {
+          logger.log(terms.map(i => {
+            return {
+              Id: i.Id,
+              Name: i.Name
+            };
+          }));
+        }
+        else {
+          const friendlyOutput: any[] = [];
+          terms.forEach(term => {
+            term.ParentTermId = '';
+            friendlyOutput.push(term);
+            if (term.Children && term.Children.length > 0) {
+              this.getFriendlyChildTerms(term, friendlyOutput);
+            }
+          });
+          logger.log(friendlyOutput.map(i => {
+            return {
+              Id: i.Id,
+              Name: i.Name,
+              ParentTermId: i.ParentTermId
+            };
+          }));
+        }
       }
     }
     catch (err: any) {
       this.handleRejectedODataJsonPromise(err);
     }
   }
+
+  private getFriendlyChildTerms(term: Term, friendlyOutput: any[] = []): any {
+    term.Children.forEach(childTerm => {
+      childTerm.ParentTermId = term.Id;
+      friendlyOutput.push(childTerm);
+      if (childTerm.Children && childTerm.Children.length > 0) {
+        this.getFriendlyChildTerms(childTerm, friendlyOutput);
+      }
+    });
+  }
+
+  private async getChildTerms(spoAdminUrl: string, res: ContextInfo, parentTerm: Term, terms: Term[] = []): Promise<void> {
```

Comment:
is `terms: Term[] = []` actually used in this method 🤔?

Reply:
Shouldn't have been there indeed. Woops!